src/AMSlib/wf/basedb.hpp

```diff
@@ -919,15 +911,13 @@ class EventBuffer
         info->orig_size)
 
     if (info->n_added > 0) {
-      // Destination buffer (of TypeValue size, either float or double)
       size_t datlen = info->n_added;  // Total number of bytes
-      int k = datlen / sizeof(TypeValue);
-      if (datlen % sizeof(TypeValue) != 0) {
-        CFATAL(RabbitMQDB,
-               false,
-               "Buffer seems corrupted or not the right type of TypeValue");
-      }
-      auto data = std::make_unique<TypeValue[]>(datlen);
+      // We get a blob of bytes here
+      auto data = std::make_unique<uint8_t[]>(datlen);
```

Comment:
Why do we get this data buffer?

Reply:
Is it necessary to copy the data into this data buffer? As you do later in the code?

`int nbyte_drained = evbuffer_remove(buffer, data.get(), datlen);`

Reply:
Yes we need that empty buffer to receive the data from `evbuffer_remove`.

Reply:
Can't you just directly publish from evbuffer_remove?

Reply:
Not really, `evbuffer` are opaque struct optimized for buffering network data. We could probably hack our way in and access the underlying pointer(s) but I am not sure it worth the trouble (libevent is using layer above low-level memory structures like [iovec](https://man7.org/linux/man-pages//man3/iovec.3type.html).

One optimization would be to add data pointers (see `evbuffer_add_reference`) to the evbuffer instead of copying data, we would save one copy per message sent.